### PR TITLE
fix: enhancement and version upgrade of google adwords offline conversion

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

resolves INT-1309

[This](https://github.com/rudderlabs/rudder-transformer/pull/2909) PR is divided into destination level changes

Google has got a new API version 15. We are updating API version of

Google adwords offline conversion

Also we are adding support of the new fields added by google for this above destination

Doc references :

offline conversion

click conversion adding consent support https://developers.google.com/google-ads/api/reference/rpc/v15/ClickConversion#consent

call conversion https://developers.google.com/google-ads/api/reference/rpc/v15/CallConversion#consent

store sales consent https://developers.google.com/google-ads/api/reference/rpc/v15/UserData#consent

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Type of change

**enhancement** 

- Existing capabilities/behavior

No change

- New capabilities/behavior

added consent field support for all of the three types of conversions

### Any technical or performance related pointers to consider with the change?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### If the PR has changes in more than 10 files, please mention why the changes were not split into multiple PRs.

N/A

### If multiple linear tasks are associated with the PR changes, please elaborate on the reason:

N/A

<hr>

### Developer checklist

- [ ] **No breaking changes are being introduced.**

- [ ] Are all related docs linked with the PR?

- [ ] Are all changes manually tested?

- [ ] Does this change require any documentation changes?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/util/index.js [hunk 1]
```diff
@@ -2108,6 +2108,28 @@ const parseConfigArray = (arr, key) => {
   return arr.map((item) => item[key]);
 };
 
+const GOOGLE_ALLOWED_CONSENT_STATUS = ['UNSPECIFIED', 'UNKNOWN', 'GRANTED', 'DENIED'];
```
Comment (koladilip) [inline src/v0/util/index.js:2111]: should this be placed in google specific utils rather than in common utils

Comment (koladilip) [reply src/v0/util/index.js:2111]: Also add unit test cases

Comment (shrouti1507) [reply src/v0/util/index.js]: moved to util and added test cases

**Review by koladilip — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**